clients/drcachesim/simulator/caching_device.h

```diff
@@ -330,6 +331,10 @@ class caching_device_t {
 
     mutable std::unique_ptr<cache_replacement_policy_t> replacement_policy_;
 
+    // For exclusive cache: Tags which previously were serviced in this cache,
+    // but moved to a child cache
+    std::unordered_set<addr_t> prev_serviced_tags_;
```

Comment:
Might be better to rename this to prev_serviced_exclusive_tags_, and also include in the comment that it is expected to be empty for inclusive caches.